### PR TITLE
Avoid rerendering the collection tree when selecting articles for edit (take 2)

### PR DESCRIPTION
**Pull request by jonathonherbert**
## What's changed?
At the moment, we rerender the entire collection tree when collection items are selected for edit. This is expensive, and you can feel the jank when the form opens on large fronts. This PR should fix the issue -- we now only render the articles that are changing state.

_NB:_ This PR corrects an issue with form saving present in the original PR for this feature.

## Implementation notes
This was a nice candidate for a refactor, as it DRYed up code across the Fronts and Clipboard components. I've also const-ified the relevant action names where necessary to allow us to depend on them outside of their reducer context, which is probably a good thing in any case!

### General
* [x]  🤖 Relevant tests added
* [x]  ✅ CI checks / tests run locally
* [x]  🔍 Checked on CODE

### Client
* [x]  🚫 No obvious console errors on the client (i.e. React dev mode errors)
* [x]  🎛️ No regressions with existing user interactions (i.e. all existing buttons, inputs etc. work)
* [ ]  📷 Screenshots / GIFs of relevant UI changes included (N/A)

### CODE REVIEW

File: client-v2/src/components/FrontsEdit/FrontDetailView.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  updateArticleFragmentMeta: (id: string, meta: ArticleFragmentMeta) => void;
+  clearArticleFragmentSelection: (id: string) => void;
+}
+
+const FrontsDetailView = ({
+  selectedArticleFragment,
+  id,
+  browsingStage,
+  clearArticleFragmentSelection,
+  updateArticleFragmentMeta
```
Comment (jonathonherbert) [inline client-v2/src/components/FrontsEdit/FrontDetailView.tsx]: I'd made a mistake here in the previous PR -- I hadn't included updateArticleFragmentMeta as a destructured prop, and as a result the component was calling the action creator, which has the same name. This meant that nothing was dispatched when onSave was called 😢

Comment (RichieAHB) [reply client-v2/src/components/FrontsEdit/FrontDetailView.tsx]: Oh wow that's fiddly ...

Comment (RichieAHB) [reply client-v2/src/components/FrontsEdit/FrontDetailView.tsx]: Is the signature exactly the same too?

Comment (jonathonherbert) [reply client-v2/src/components/FrontsEdit/FrontDetailView.tsx]: yuh-huh 👎 Potential way of spotting these things in future -- adopt naming convention for actions, e.g. actionUpdateArticleFragmentMeta

**Pull request merged**